src/Interpreters/InstrumentationProfilingLog.cpp

```diff
@@ -0,0 +1,68 @@
+#include <Interpreters/InstrumentationProfilingLog.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <Columns/IColumn.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDateTime64.h>
+
+namespace DB
+{
+
+NamesAndTypesList InstrumentationProfilingLogElement::getNamesAndTypes()
+{
+    return NamesAndTypesList
+    {
+        {"name", std::make_shared<DataTypeString>()},
+        {"cat", std::make_shared<DataTypeString>()},
+        {"ph", std::make_shared<DataTypeString>()},
+        {"pid", std::make_shared<DataTypeUInt64>()},
+        {"tid", std::make_shared<DataTypeUInt64>()},
+        {"ts", std::make_shared<DataTypeDateTime64>(6)},
+        {"event_time", std::make_shared<DataTypeDateTime>()},
+        {"query_id", std::make_shared<DataTypeString>()},
+        {"function_id", std::make_shared<DataTypeInt32>()},
+    };
+}
+
+ColumnsDescription InstrumentationProfilingLogElement::getColumnsDescription()
+{
+    return ColumnsDescription
+    {
+        {"name", std::make_shared<DataTypeString>(), "Name of the instrumented function."},
+        {"cat", std::make_shared<DataTypeString>(), "Category of the instrumented function."},
+        {"ph", std::make_shared<DataTypeString>(), "Phase: entry or exit."},
+        {"pid", std::make_shared<DataTypeUInt64>(), "Process ID."},
```

Comment:
I think we don't need ot do a 1:1 mapping of everything we need for the Chrome trace format. I think there's no need to save extra information if we can build those parameters based on some others:

- The PID is going to be the same
- Why do we need to save an entry and then an exit `phase`? Why don't we store the timestamp where something started + duration and then we build the two Chrome trace format rows from there? That way we avoid having 2 entries when we could have only 1. It's also easier to understand by a human in my opinion, cause the the whole event info is in a single row.
- We can use whatever in the `cat` part as far as I know, right?